core/src/main/java/com/scalar/db/storage/common/AbstractDistributedStorage.java

```diff
@@ -0,0 +1,68 @@
+package com.scalar.db.storage.common;
+
+import com.scalar.db.api.Delete;
+import com.scalar.db.api.DistributedStorage;
+import com.scalar.db.api.Get;
+import com.scalar.db.api.Mutation;
+import com.scalar.db.api.Put;
+import com.scalar.db.api.Scan;
+import com.scalar.db.util.ScalarDbUtils;
+import java.util.List;
+import java.util.Optional;
+
+public abstract class AbstractDistributedStorage implements DistributedStorage {
```

Comment:
This PR introduces AbstractDistributedStorage that extracts the common logic of the DistributedStorage implementations.